File: PROGRAMMERS/LEVEL1/같은 숫자는 싫어.cpp
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr)
{
    vector<int> answer;
    answer.push_back(arr.back());
    arr.pop_back();
    while (!arr.empty()) {
        if (arr.back() != answer.back()) {
            answer.push_back(arr.back());
        }
        arr.pop_back();
    }
    reverse(answer.begin(), answer.end());
    return answer;
}

File: PROGRAMMERS/LEVEL1/같은 숫자는 싫어.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Solution {
    public Stack<Integer> solution(int []arr) {
        
    	Stack<Integer> answer = new Stack<Integer>();
    	
    	answer.push(arr[0]);
    	for(int i=1;i<arr.length;i++) {
    		if(answer.peek()==arr[i]) continue;
    		answer.push(arr[i]);
    	}
        

        return answer;
    }
}